ReactAndroid/src/main/java/com/facebook/react/modules/network/NetworkingModule.java

```diff
@@ -189,7 +189,22 @@ public void onResponse(Response response) throws IOException {
               callback.invoke(0, null, e.getMessage());
               return;
             }
-            callback.invoke(response.code(), null, responseBody);
+
+            WritableMap responseHeaders = Arguments.createMap();
+            Headers headers = response.headers();
+            for (int i = 0; i < headers.size(); i++) {
+              // multiple values for the same header
+              if (responseHeaders.hasKey(headers.name(i))) {
+                responseHeaders.putString(
+                  headers.name(i),
+                  responseHeaders.getString(headers.name(i) + "," + headers.value(i))
```

Comment:
This doesn't look correct: you're getting the string out of the responseHeaders map corresponding to the key `{header_key},{header_value}`. Was this tested?